tests/lib/rules/prefer-named-capture-group.js

```diff
@@ -104,15 +124,53 @@ ruleTester.run("prefer-named-capture-group", rule, {
                 data: { group: "([0-9]{4})" },
                 line: 1,
                 column: 1,
-                endColumn: 21
+                endColumn: 21,
+                suggestions: [
+                    {
+                        messageId: "addGroupName",
+                        output: "RegExp('(?<temp1>[0-9]{4})')"
+                    },
+                    {
+                        messageId: "addNonCapture",
+                        output: "RegExp('(?:[0-9]{4})')"
+                    }
+                ]
             }]
         },
         {
             code: "new RegExp(`a(bc)d`)",
             errors: [{
                 messageId: "required",
                 type: "NewExpression",
-                data: { group: "(bc)" }
+                data: { group: "(bc)" },
+                suggestions: [
+                    {
+                        messageId: "addGroupName",
+                        output: "new RegExp(`a(?<temp1>bc)d`)"
+                    },
+                    {
+                        messageId: "addNonCapture",
+                        output: "new RegExp(`a(?:bc)d`)"
+                    }
+                ]
+            }]
+        },
+        {
+            code: "new RegExp('\u1234\u5678(?:a)(b)');",
```

Comment:
`\u1234` and `\u5678` are evaluated for the `code` string, so this tests the following source code:

```js
new RegExp('ሴ噸(?:a)(b)');
```

If we want to test `new RegExp("\u1234\u5678(?:a)(b)");` source code, we should escape `\`:

```suggestion
            code: "new RegExp('\\u1234\\u5678(?:a)(b)');",
```

Reply:
Got it, thanks!